imap_processing/codice/tests/test_decom.py

```diff
@@ -43,16 +46,78 @@ def validation_data():
         f"{imap_module_directory}/codice/tests/data/"
         f"idle_export_raw.COD_NHK_20230822_122700.csv"
     )
-    validation_data = pd.read_csv(validation_file, index_col="SHCOARSE")
+    raw_data = pd.read_csv(validation_file, index_col="SHCOARSE")
     # Remove the timestamp column and data
-    if "timestamp" in validation_data.columns:
-        validation_data.drop(columns=["timestamp"], inplace=True, errors="ignore")
+    if "timestamp" in raw_data.columns:
+        raw_data.drop(columns=["timestamp"], inplace=True, errors="ignore")
+
+    dataset = xr.Dataset()
+    dataset["First_mnemonic"] = xr.DataArray(np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
 
     # Return the data
-    return validation_data
+    return raw_data
```

Comment:
you could do something similar to what I did here `create_dataset()`. If you look at references, you can do same thing here. Its input is packet list which you already have from `decom_test_data()`. `create_dataset()` gives back `xr.Dataset` which you can then use as input to `convert_raw_to_eu()`.

https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/swe/utils/swe_utils.py